src/main/java/com/salesforce/mirus/MirusSourceTask.java

```diff
@@ -181,9 +188,11 @@ private void seekToOffsets(List<TopicPartition> partitionIds) {
 
     try {
       logger.trace("Calling poll");
+      checkCommitFailure();
       ConsumerRecords<byte[], byte[]> result = consumer.poll(consumerPollTimeoutMillis);
       logger.trace("Got {} records", result.count());
       if (!result.isEmpty()) {
+        setPollTime();
```

Comment:
I think it this would be easier to understand if we simply set `lastNewRecordTime=time.milliseconds()`here, then in `checkCommitFailure` we simply look for  `lastNewRecordTime - commitTime > commitFailureRestartMs`. So we fail if the time between the most recent record arrival and most recent successful commit is more than `commitFailureRestartMs`.

Reply:
In this `setPollTime` method I also try to handle the case you mentioned previously. Thus extract the `setCommitTime` method although it's only one line call.

> I'm wondering what will happen when there is no new data for a long time, and then a sudden burst of continuous data.

Reply:
Thanks @YongGang. Yes, I can see that you've addressed the case I mentioned previously, but I think there is an also an opportunity here to simplify things a little.

Reply:
Thanks Paul, updated as you suggested.